src/main/java/carpet/script/value/ScreenHandlerValue.java

```diff
@@ -0,0 +1,380 @@
+            case "secondary_effect" -> getPropertyForType(BeaconScreenHandler.class, "beacon", 2, propertyName);
+
+            case "brew_time" -> getPropertyForType(BrewingStandScreenHandler.class, "brewing_stand", 0, propertyName);
+            case "brewing_fuel" -> getPropertyForType(BrewingStandScreenHandler.class, "brewing_stand", 1, propertyName);
+
+            case "enchantment_power_1" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 0, propertyName);
+            case "enchantment_power_2" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 1, propertyName);
+            case "enchantment_power_3" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 2, propertyName);
+            case "enchantment_seed" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 3, propertyName);
+            case "enchantment_id_1" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 4, propertyName);
+            case "enchantment_id_2" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 5, propertyName);
+            case "enchantment_id_3" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 6, propertyName);
+            case "enchantment_level_1" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 7, propertyName);
+            case "enchantment_level_2" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 8, propertyName);
+            case "enchantment_level_3" -> getPropertyForType(EnchantmentScreenHandler.class, "enchantment", 9, propertyName);
+
+            case "banner_pattern" -> getPropertyForType(LoomScreenHandler.class, "loom", 0, propertyName);
+
+            case "stonecutter_recipe" -> getPropertyForType(StonecutterScreenHandler.class, "stonecutter", 0, propertyName);
+
+            default -> throw new InternalExpressionException("Invalid screen handler property: " + propertyName);
+        };
+
+    }
+
+    public Value queryProperty(String propertyName) {
+        if(propertyName.equals("name")) return FormattedTextValue.of(this.name);
+        if(propertyName.equals("open")) return BooleanValue.of(this.isOpen());
+        Property property = getProperty(propertyName);
+        return NumericValue.of(property.get());
+    }
+
+    public Value modifyProperty(String propertyName, List<Value> lv) {
+        Property property = getProperty(propertyName);
+        int intValue = NumericValue.asNumber(lv.get(0)).getInt();
+        property.set(intValue);
+        this.screenHandler.syncState();
+        return Value.TRUE;
+    }
+
+    @Override
+    public String getString() {
+        return this.typestring + "_screen_handler";
+    }
+
+    @Override
+    public boolean getBoolean() {
+        return true;
+    }
+
+    @Override
+    public String getTypeString()
+    {
+        return "screen_handler";
```

Comment:
Yeah, simplify in names and docs to 'screen' especially that you have `create_screen` not `create_screen_handler`. There shall not be other screens.